### PR TITLE
AM-782:Spike for Integration test

### PR DESCRIPTION
**Pull request by nitinsahu1**
AM-782:Spike for Integration test

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-782

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/WelcomeControllerIntegrationTest.java [hunk 1]
```diff
@@ -57,5 +60,26 @@ public void welComeAPITest() throws Exception {
         assertEquals(
             "Service is unavailable:", 200, 200);
```
Comment (aashishaggarwal123) [inline src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/WelcomeControllerIntegrationTest.java:61]: the message is looking wrong for 200 status

File: src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/WelcomeControllerIntegrationTest.java [hunk 2]
```diff
@@ -57,5 +60,26 @@ public void welComeAPITest() throws Exception {
...
     }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {"classpath:sql/insert_role_assignment_request.sql"})
+    public void shoudGetRecordCountFromRequestTable() throws Exception {
+        final int count = template.queryForObject(COUNT_RECORDS, Integer.class);
+        logger.info(" Total number of records fetched from role assignment request table...{}", count);
+        assertEquals(
+            "role_assignment_request record count ", 5, count);
```
Comment (aashishaggarwal123) [inline src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/WelcomeControllerIntegrationTest.java:70]: would this work in the PR and get 5 records?

File: src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/WireMockBaseTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        final String hostUrl = "http://localhost:" + wiremockPort;
+
+        LOG.info("Wire mock test, host url is {}", hostUrl);
+
+        ReflectionTestUtils.setField("caseDefinitionHost", "caseDefinitionHost", hostUrl);
+        ReflectionTestUtils.setField("uiDefinitionHost", "uiDefinitionHost", hostUrl);
+        ReflectionTestUtils.setField("idamHost", "idamHost", hostUrl);
+        ReflectionTestUtils.setField("userProfileHost", "userProfileHost", hostUrl);
+        ReflectionTestUtils.setField("draftHost", "draftHost", hostUrl);
+    }
```
Comment (aashishaggarwal123) [inline src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/WireMockBaseTest.java:33]: do we need to set these fields?

File: src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/WelcomeControllerIntegrationTest.java (hunk 1 above)
Comment (nitinsahu1) [reply src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/WelcomeControllerIntegrationTest.java]: Message updated.

File: src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/WelcomeControllerIntegrationTest.java (hunk 2 above)
Comment (nitinsahu1) [reply src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/WelcomeControllerIntegrationTest.java:68]: its embedded postgress database created on the fly and perform db operations. This test passed successfully in PR.

File: src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/WireMockBaseTest.java (hunk 3 above)
Comment (nitinsahu1) [reply src/integrationTest/java/uk/gov/hmcts/reform/roleassignment/controller/WireMockBaseTest.java:33]: This is a sample fetched from ccd store and would be changed later once we will integrate it with our APIs.

**Pull request merged**